pkg/apis/secret/v0alpha1/exposed_secure_value.go

```diff
@@ -28,13 +23,16 @@ var (
 
 // NewExposedSecureValue creates a new exposed secure value wrapper.
 func NewExposedSecureValue(v string) ExposedSecureValue {
-	return ExposedSecureValue{v: v}
+	return ExposedSecureValue(v)
 }
 
-// DangerouslyExposeDecryptedValue will return the decrypted secure value.
+// DangerouslyExposeAndConsumeValue will move the decrypted secure value out of the wrapper and return it.
+// Further attemps to call this method will simply return an empty string.
 // The function name is intentionally kept long and weird because this is a dangerous operation and should be used carefully!
-func (s ExposedSecureValue) DangerouslyExposeDecryptedValue() string {
-	return s.v
+func (s *ExposedSecureValue) DangerouslyExposeAndConsumeValue() string {
+	tmp := *s
```

Comment:
Have you considered asserting `s` is not empty? One more check to ensure it isn't called more than once

Reply:
@PoorlyDefinedBehaviour do you mean to change this to return an `error`?

Reply:
I mean panicking